docs/source/processing.rst

```diff
@@ -491,4 +491,53 @@ Both :obj:`datasets.Dataset` and :obj:`datasets.DatasetDict` objects can be save
 Controling the cache behavior
 -----------------------------------
 
-[UNDER CONSTRUCTION]
+When applying transforms on a dataset, the data are stored in cache files.
+The caching mechanism allows to reload an existing cache file if it's already been computed.
+
+Reloading a dataset is possible since the cache files are named using the dataset fingerprint, which is updated after each transform.
+
+
+Fingerprinting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The fingerprint of a dataset in a given state is an internal value computed by combining the fingerprint of the previous state and a hash of the latest transform that was applied.
+The initial fingerprint is computed using a hash of the arrow table, or a hash of the arrow files if the dataset lives on disk.
+
+For example:
+
+.. code-block::
+
+    >>> from datasets import Dataset
+    >>> dataset1 = Dataset.from_dict({"a": [0, 1, 2]})
+    >>> dataset2 = dataset1.map(lambda x: {"a": x["a"] + 1})
+    >>> print(dataset1._fingerprint, dataset2._fingerprint)
+    d19493523d95e2dc 5b86abacd4b42434
+
+The new fingerprint is a combination of the previous fingerprint and the hash of the given transform. For a transform to be hashable, it needs to be picklable using dill or pickle.
```

Comment:
```suggestion
The new fingerprint is a combination of the previous fingerprint and the hash of the given transform. For a transform to be hashable, it needs to be picklable using dill or pickle. In particular for :func:`datasets.Dataset.map`, you need to provide a picklable processing method to apply on the dataset so that a determinist fingerprint can be computed by hashing the full state of the provided method (the fingerprint is computed taking into account all the dependencies of the method you provide). 
```